### PR TITLE
Zahid

### PR DESCRIPTION
**Pull request by ztabani**
1- Implemented settings validation for Prime Clinical Settings. 
2- Implemented sync functionality in windows service
3- Working on windows service controller screen for Install/Unistall/Start/Stop service features as part of RnD work.

### CODE REVIEW

File: curogram-integration-core/Helpers/LogHelper.cs [hunk 1]
```diff
@@ -53,110 +51,153 @@ public static void WriteToFile(string logMessage)
...
             string logFile = String.Empty;
             StreamWriter logWriter;
 
-            string logFolderPath = Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "Logs");
+            logFile = Path.Combine(ErrorLogPath, DateTime.Today.ToString("yyyy-MM-dd").ToString() + ".txt");
+            if (!Directory.Exists(ErrorLogPath))
+                Directory.CreateDirectory(ErrorLogPath);
+            try
+            {
+                if (File.Exists(logFile))
```
Comment (nafisi442) [inline curogram-integration-core/Helpers/LogHelper.cs]: this doesn't seem why it should be in a try/catch block. what part of it is potential for error? only that part should be in try/catch block. please maintain this principal for all try/catch blocks

File: curogram-integration-wingui/frmSettings.cs [hunk 2]
```diff
@@ -20,7 +21,7 @@ public partial class frmSettings : Form
         public frmSettings()
         {
             InitializeComponent();
-            syncSettings = Utils.LoadSyncSettings();
+            syncSettings = LogHelper.LoadSyncSettings();
```
Comment (nafisi442) [inline curogram-integration-wingui/frmSettings.cs]: why does the sync settings come from the log helper? it doesn't sound, right?

Comment (ztabani) [reply curogram-integration-wingui/frmSettings.cs]: Yes, its wrong, my mistake .. Updating in next PR.

File: curogram-integration-core/Helpers/LogHelper.cs (hunk 1 above)
Comment (ztabani) [reply curogram-integration-core/Helpers/LogHelper.cs]: Noted.

**Pull request merged**